### PR TITLE
Support translating YearSchedule Begin/End dates to ScheduleRules

### PR DESCRIPTION
**Pull request by eringold**
Adds parsing of multiple YearSchedules with BeginDate and EndDate fields and translation into ScheduleRules. 

Close https://github.com/NREL/gbxml-to-openstudio/issues/131, Jira revit 213603.

---

> The [Schedule](https://www.gbxml.org/schema/7-03/GreenBuildingXML_Ver7.03.xsd#element_Schedule) element can have 1 or more [YearSchedule](https://www.gbxml.org/schema/7-03/GreenBuildingXML_Ver7.03.xsd#element_YearSchedule) with BeginDate and EndDate to tell when it takes effect.
> 
> - [x] However, the current measures only take the first year: https://github.com/NREL/gbxml-to-openstudio/blob/9868a5cb763ff6de1ffdf31a5aaabbbf973f3fc6/measures/advanced_import_gbxml/measure.rb#L325
> 
> - [x] The measures then hardcode the full year range, regardless of the BeginDate and EndDate: https://github.com/NREL/gbxml-to-openstudio/blob/9868a5cb763ff6de1ffdf31a5aaabbbf973f3fc6/measures/advanced_import_gbxml/resources/os_lib_adv_import.rb#L831
> 
> - [x] The existing logic that sets the schedule based on building type should be removed: https://github.com/NREL/gbxml-to-openstudio/blob/9868a5cb763ff6de1ffdf31a5aaabbbf973f3fc6/measures/advanced_import_gbxml/resources/os_lib_adv_import.rb#L869-L899
>

### CODE REVIEW

File: measures/advanced_import_gbxml/measure.rb [hunk 1]
```diff
@@ -321,9 +321,18 @@ def run(model, runner, user_arguments)
...
+
+      # get Year Schedule info (multiple per Schedule)
+      year_schedules = {}
+      element.elements.each('YearSchedule') do |year|
+        id = year.attributes['id']
+        sch_week = year.elements['WeekScheduleId'].attributes['weekScheduleIdRef']
+        begin_date = year.elements['BeginDate'].text
+        end_date = year.elements['EndDate'].text
+        year_schedules[id] = {'sch_week' => sch_week, 'begin_date' => begin_date, 'end_date' => end_date}
+      end
```
Comment (eringold) [inline measures/advanced_import_gbxml/measure.rb:334]: get YearSchedule info as nested hash in the :schedules hash

File: measures/advanced_import_gbxml/resources/os_lib_adv_import.rb [hunk 2]
```diff
@@ -819,92 +820,94 @@ def self.import_schs(runner, model, building_type, schedules, week_schedules, da
...
       # get schedule name
       if schedule_data['name'].nil?
         ruleset_name = id
       else
         ruleset_name = schedule_data['name']
       end
-      date_range = '1/1-12/31'
-      # winter_design_day = nil
-      # summer_design_day = nil
+      # date_range = '1/1-12/31'
```
Comment (eringold) [inline measures/advanced_import_gbxml/resources/os_lib_adv_import.rb:836]: remove hard-coded `date_range`

File: measures/advanced_import_gbxml/resources/os_lib_adv_import.rb [hunk 3]
```diff
@@ -819,92 +820,94 @@ def self.import_schs(runner, model, building_type, schedules, week_schedules, da
...
+            days = 'Sat/Sun'
+          when 'WeekendOrHoliday'
+            days = 'Sat/Sun/Hol'
+          when 'Custom'
+            days = 'Custom' # TODO
+          when 'All' 
+            days = 'Mon/Tue/Wed/Thu/Fri/Sat/Sun'
+          else
+            days = day_type
+          end
```
Comment (eringold) [inline measures/advanced_import_gbxml/resources/os_lib_adv_import.rb:897]: This section replaces the hard-coded day types applied by the BuildingTypeHelper class in /resources/building_type_helper.rb. Not sure if that helper was needed for special schedule parsing, can re-enable if so. Leaving the resources file in the measure for now.

File: measures/advanced_import_gbxml/resources/os_lib_schedules.rb [hunk 4]
```diff
@@ -539,17 +539,36 @@ def self.createComplexSchedule(model, options = {})
...
+      holiday_day = OpenStudio::Model::ScheduleDay.new(model)
+      sch_ruleset.setHolidaySchedule(holiday_day)
+      holiday_day = sch_ruleset.holidaySchedule
+      holiday_day.setName("#{sch_ruleset.name} Holiday Day")
+      options['holiday_day'].each do |data_pair|
+        hour = data_pair[0].truncate
+        min = ((data_pair[0] - hour) * 60).to_i
+        holiday_day.addValue(OpenStudio::Time.new(0, hour, min, 0), data_pair[1])
+      end
+    end
```
Comment (eringold) [inline measures/advanced_import_gbxml/resources/os_lib_schedules.rb:565]: Process default days and holidays

**Review by MatthewSteen — APPROVED** [review]
**Before**
ScheduleRules set by BuildingTypeHelper.

image

**After**
ScheduleRules set by gbXML YearSchedules.

image

**Pull request merged**